### PR TITLE
DDFNEXT-220: Upgrade Storybook and Prettier

### PR DESCRIPTION
**Pull request by JacobArrow**
#### Link to issue

https://reload.atlassian.net/browse/DDFBRA-161

#### Description

- Upgraded Storybook and its dependencies 
- Migrated Storybook config to match breaking changes
- Migrated stories with `npx storybook@latest migrate`
- Migrated `.mdx` files with `npx storybook@latest migrate`
- Rewrote all stories to use `args: {}` instead of `argTypes.defaultValue: {}`
- Upgraded Prettier and Stylelint packages to work with `@whitespace/storybook-addon-html`
- Fixed various breaking changes and migration errors in stories

**Beware that the newest version of Storybook introduces some new React warnings while compiling.**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -69,28 +71,28 @@
...
+    "eslint-plugin-prettier": "^5.0.0",
     "eslint-plugin-react": "^7.37.2",
     "eslint-plugin-react-hooks": "^4.6.2",
     "eslint-webpack-plugin": "^4.2.0",
     "flatpickr": "^4.6.13",
     "json": "^11.0.0",
     "markdownlint-cli2": "^0.4.0",
     "postcss": "^8.4.47",
-    "prettier": "^2.7.1",
+    "prettier": "^3.3.3",
```
Comment (JacobArrow) [inline package.json:82]: We need to upgrade Prettier to make [storybook-addon-html](https://github.com/whitespace-se/storybook-addon-html?tab=readme-ov-file#requirements) work with Storybook 8, which is why Stylelint also has been upgraded.

File: package.json [hunk 2]
```diff
@@ -77,15 +79,16 @@
...
     "react": "^17.0.2",
     "react-docgen-typescript": "^2.1.0",
     "react-dom": "^17.0.2",
     "react-helmet": "^6.1.0",
-    "react-scripts": "^4.0.3",
+    "react-scripts": "^5.0.0",
+    "react-syntax-highlighter": "^15.6.1",
```
Comment (kasperg) [inline package.json:88]: This seems like an addition. Next time I would appreciate a mention in the commit about why this is necessary - a bit like what you did for `prettier`.

File: src/stories/Blocks/header/Header.tsx [hunk 3]
```diff
@@ -29,10 +29,7 @@ export const Header = ({
 }: HeaderProps) => {
   const [isDropdownOpen, setIsDropdownOpen] = useState(openDropdown);
 
-  useEffect(() => {
-    require("./header-sticky");
-    require("./header-state");
-  }, []);
```
Comment (kasperg) [inline src/stories/Blocks/header/Header.tsx]: As mentioned previously I hope this gets reintroduced later on.

File: src/stories/Library/card-list-page/SearchResultPage.stories.tsx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { SearchResultPage } from "./SearchResultPage";
 import { SearchResultPageSkeleton } from "./SearchResultPageSkeleton";
 
 export default {
-  title: "Blocks / Search Result Page",
+  title: "Blocks / Search Result Page Skeleton",
```
Comment (kasperg) [inline src/stories/Library/card-list-page/SearchResultPage.stories.tsx]: We have discussed whether these new titles make sense. I hope to see this addressed in a later commit.

**Review by kasperg — APPROVED** [review]

**Pull request merged**